WcaOnRails/app/models/registration.rb

```diff
@@ -102,7 +107,7 @@ def waiting_list_info
 
   validate :must_register_for_gte_one_event
   private def must_register_for_gte_one_event
-    if events.empty?
+    if registration_events.length { |r| !r.marked_for_destruction? } == 0
```

Comment:
Oh ok, so the only way it could be marked for destruction is through the competition controller, right?
Is it why we actually use `.empty?` [here](https://github.com/cubing/worldcubeassociation.org/pull/759/files/092c6c899303fd543fc8b55597e4fb8ebc2c16fe#diff-848ffc153c457b964c6c8a9d204cea61R98) instead of a similar filter?